File: Data Structures & Algorithms/is-anagram/submission-1.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        mp = {}

        for i in s:
            if mp.get(i) is None:
                mp[i] = 0
            mp[i] += 1

        for i in t:
            if mp.get(i) is None:
                return False
            mp[i] -= 1

        for k in mp.values():
            if k != 0:
                return False
        return True        

        

File: Data Structures & Algorithms/is-palindrome/submission-1.java
class Solution {
    public boolean isPalindrome(String s) {
        int j = s.length() - 1;
        int i = 0;

        while(i<j){
            if(!Character.isLetterOrDigit(s.charAt(i))){
                i++;
                continue;
            }
            if(!Character.isLetterOrDigit(s.charAt(j))){
                j--;
                continue;
            }
            if(Character.toLowerCase(s.charAt(i)) != Character.toLowerCase(s.charAt(j))){
                return false;
            }
            i++;
            j--;
        }
        return true;
    }
}
